api/v1alpha1/kongplugin_installation_types.go

```diff
@@ -0,0 +1,129 @@
+	Image string `json:"image"`
+
+	// SecretRef allows referring specific Kubernetes Secret to use for OCI registry authentication for pulling an image with Custom Kong Plugin.
+	// The Secret format should follow https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry.
+	// When the field is omitted it is assumed that the image is public and can be fetched without providing credentials.
+	//
+	//+optional
+	SecretRef *corev1.SecretReference `json:"secretRef,omitempty"`
+}
+
+// KongPluginInstallationStatus defines the observed state of KongPluginInstallation.
+type KongPluginInstallationStatus struct {
+	// Conditions describe the current conditions of this KongPluginInstallation.
+	//
+	//+listType=map
+	//+listMapKey=type
+	//+kubebuilder:validation:MaxItems=8
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// KongPluginInstallationConditionType is the type for status conditions on
+// KongPluginInstallation resources. This type should be used with the
+// KongPluginInstallationStatus.Conditions field.
+type KongPluginInstallationConditionType string
+
+// KongPluginInstallationConditionReason defines the set of reasons that explain why
+// a particular KongPluginInstallation condition type has been raised.
+type KongPluginInstallationConditionReason string
+
+const (
+	// This condition indicates whether the controller has fetched the plugin image
+	// and made it available for use as a specific Custom Kong Plugin.
+	//
+	// It is a positive-polarity summary condition, and so should always be
+	// present on the resource with ObservedGeneration set.
+	//
+	// It should be set to Unknown if the controller performs updates to the
+	// status before it has all the information it needs to be able to determine
+	// if the condition is true.
+	//
+	// Possible reasons for this condition to be "True" are:
+	//
+	// * "Ready"
+	//
+	// Possible reasons for this condition to be "False" are:
+	//
+	// * "Failed"
+	// * "Pending"
+	//
+	// Possible reasons for this condition to be "Unknown" are:
+	//
+	// * "Pending".
+	//
+	KongPluginInstallationConditionStatusAccepted KongPluginInstallationConditionType = "Ready"
+
+	// KongPluginInstallationReasonReady is used with the "Ready" condition when
+	// the condition is "True".
+	KongPluginInstallationReasonReady KongPluginInstallationConditionReason = "Ready"
+
+	// KongPluginInstallationReasonFailed is used with the "Ready" condition when
+	// the KongPluginInstallation can't be configured e.g. image can't be fetched.
+	// More details can be obtained from the condition's message.
+	KongPluginInstallationReasonFailed KongPluginInstallationConditionReason = "Failed"
+
+	// KongPluginInstallationReasonPending is used with the "Ready" condition when the requested
+	// controller has started processing the KongPluginInstallation, but it hasn't finished yet.
+	KongPluginInstallationReasonPending KongPluginInstallationConditionReason = "Pending"
```

Comment:
Using `Pending` as a reason when the controller has already picked this for reconciliation is slightly semantically misleading.

WDYT about following Gateway API in this aspect and using that reason when the controller is yet to pick this resource for reconciliation (that means setting this reason with `Unknown` status as default)?

ref: https://github.com/kubernetes-sigs/gateway-api/blob/c1b375bda97d344948a50ebd40814ffcd68b78ac/apis/v1/gateway_types.go#L858-L861

---

OTOH if we want to express the resource state when the controller has picked this up - or has "accepted" it - I suggest we consider adding a separate condition type for this and call it `Accepted` https://github.com/kubernetes-sigs/gateway-api/blob/c1b375bda97d344948a50ebd40814ffcd68b78ac/apis/v1/gateway_types.go#L820-L845.

WDYT?